### PR TITLE
Flex size support

### PR DESCRIPTION
**Pull request by lievenjanssen**
Currently I generate `flex: 1 1 $value;` where `$value` is the flex size. I was wondering if this is the correct approach or if I should just set the flex-basis like this `flex-basis: $value;`. What do you think?

### CODE REVIEW

**Comment by philmtd** [comment]
Thank you (again) for this PR!

> Currently I generate `flex: 1 1 $value;` where `$value` is the flex size. I was wondering if this is the correct approach or if I should just set the flex-basis like this `flex-basis: $value;`. What do you think?

I think this is correct and would work according to the defaults of fxFlex in Flex Layout (checking here https://github.com/angular/flex-layout/wiki/fxFlex-API and here https://github.com/angular/flex-layout/blob/master/projects/libs/flex-layout/flex/flex/flex.ts).

So I think this implementation should be fine.

**Review by philmtd — COMMENTED** [review]
I have one suggestion (twice). Could you check that? Otherwise this looks good to me.

And before I can merge it, you'd need to kindly resolve the conflicts that happend due to the other already merged PRs 🙂

File: src/lib/flex-size-attributes.scss [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/// @param {number} $from
+///   Will start generating selectors from $from
+/// @param {number} $to
+///   Will generate selectors until including $to
+/// @param {string} $increment
+///   Will generate selectors in increments of $increment
+/// @param {bool} $includeSelectorsForMediaSizes
+///	  Will generate additional selectors for flex layout media sizes
+@mixin flex-size-attributes($from: 5, $to: 100, $increment: 5, $includeSelectorsForMediaSizes: true) {
+    @if math.div($from, $increment) != math.round(math.div($from, $increment)) or math.div($to, $increment) != math.round(math.div($to, $increment)) {
```
[the remark this answers is not in the document]
Comment (lievenjanssen) [reply src/lib/flex-size-attributes.scss]: Done

File: src/lib/flex-size-classes.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+/// @param {number} $from
+///   Will start generating selectors from $from
+/// @param {number} $to
+///   Will generate selectors until including $to
+/// @param {string} $increment
+///   Will generate selectors in increments of $increment
+/// @param {bool} $includeSelectorsForMediaSizes
+///	  Will generate additional selectors for flex layout media sizes
+@mixin flex-size-classes($from: 5, $to: 100, $increment: 5, $includeSelectorsForMediaSizes: true) {
+    @if math.div($from, $increment) != math.round(math.div($from, $increment)) or math.div($to, $increment) != math.round(math.div($to, $increment)) {
```
[the remark this answers is not in the document]
Comment (lievenjanssen) [reply src/lib/flex-size-classes.scss]: Done

**Comment by lievenjanssen** [comment]
> you'd need to kindly resolve the conflicts that happened due to the other already merged PRs

Fixed. This is ready to be merged I think.

**Review by philmtd — APPROVED** [review]

**Comment by philmtd** [comment]
Thanks! Looks good to me. I'll try to document and release everything in the next days. Is that okay with you, or do you need it asap?

**Comment by lievenjanssen** [comment]
> Thanks! Looks good to me. I'll try to document and release everything in the next days. Is that okay with you, or do you need it asap?

I suggest to still do the increment change for gap selectors before we do a major version release. I would want to use it asap but I can temporarily work around that.

**Pull request merged**